homeassistant/helpers/device_registry.py

```diff
@@ -813,6 +814,9 @@ def async_update_device(
 
         config_entries = old.config_entries
 
+        if merge_connections is not UNDEFINED and new_connections is not UNDEFINED:
+            raise HomeAssistantError
```

Comment:
Can we add a message. 
```suggestion
            raise HomeAssistantError("Cannot define both merge_connections and new_connections")
```